AsyncDisplayKit/ASImageNode.mm

```diff
@@ -153,88 +147,104 @@ - (NSObject *)drawParametersForAsyncLayer:(_ASDisplayLayer *)layer;
 {
   BOOL hasValidCropBounds = _cropEnabled && !CGRectIsNull(_cropDisplayBounds) && !CGRectIsEmpty(_cropDisplayBounds);
 
-  return [[_ASImageNodeDrawParameters alloc] initWithCrop:_cropEnabled
-                                                   opaque:self.opaque
-                                                    image:self.image
-                                                   bounds:(hasValidCropBounds ? _cropDisplayBounds : self.bounds)
-                                            contentsScale:self.contentsScaleForDisplay
-                                          backgroundColor:self.backgroundColor
-                                              contentMode:self.contentMode
-                                                 cropRect:self.cropRect
-                                   imageModificationBlock:self.imageModificationBlock];
+  return [[_ASImageNodeDrawParameters alloc] initWithBounds:self.bounds
+                                                     opaque:self.opaque
+                                              contentsScale:self.contentsScaleForDisplay
+                                            backgroundColor:self.backgroundColor
+                                                contentMode:self.contentMode];
 }
 
-+ (UIImage *)displayWithParameters:(_ASImageNodeDrawParameters *)parameters isCancelled:(asdisplaynode_iscancelled_block_t)isCancelled
+- (UIImage *)displayWithParameters:(_ASImageNodeDrawParameters *)parameters isCancelled:(asdisplaynode_iscancelled_block_t)isCancelled
 {
-  UIImage *image = parameters.image;
-
+  ASDN::MutexLocker l(_imageLock);
```

Comment:
I *think* I've done that. the other parameters that are accessed after this are all now protected by _imageLock. I also have the unlock after.